2024-04-19 10:47:29,202 INFO: Tinker Full-stack App [in /mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/__init__.py:61]
2024-04-19 11:06:17,341 INFO: Tinker Full-stack App [in /mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/__init__.py:61]
2024-04-19 16:26:44,560 INFO: Tinker Full-stack App [in /mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/__init__.py:61]
2024-05-11 08:45:54,220 INFO: Tinker Full-stack App [in /mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/__init__.py:61]
2024-05-11 08:46:32,758 INFO: Tinker Full-stack App [in /mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/__init__.py:61]
2024-05-11 08:46:49,411 ERROR: Exception on /login [GET] [in /home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py:825]
Traceback (most recent call last):
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/routes.py", line 153, in login
    return render_template('login.html', title='Login', form=form)
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/templating.py", line 152, in render_template
    return _render(app, template, context)
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/templating.py", line 133, in _render
    rv = template.render(context)
  File "/home/missmandazi/.local/lib/python3.8/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/home/missmandazi/.local/lib/python3.8/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/templates/login.html", line 1, in top-level template code
    {% extends 'base.html' %}
  File "/mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/templates/base.html", line 52, in top-level template code
    {% block content %}
  File "/mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/templates/login.html", line 37, in block 'content'
    <a href="{{ url_for(request_password_reset) }}">Reset Here</a>
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py", line 1018, in url_for
    if endpoint[:1] == ".":
jinja2.exceptions.UndefinedError: 'request_password_reset' is undefined
2024-05-11 08:49:09,493 INFO: Tinker Full-stack App [in /mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/__init__.py:61]
2024-05-11 08:49:52,525 ERROR: Exception on /login [GET] [in /home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py:825]
Traceback (most recent call last):
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/routes.py", line 153, in login
    return render_template('login.html', title='Login', form=form)
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/templating.py", line 152, in render_template
    return _render(app, template, context)
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/templating.py", line 133, in _render
    rv = template.render(context)
  File "/home/missmandazi/.local/lib/python3.8/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/home/missmandazi/.local/lib/python3.8/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/templates/login.html", line 1, in top-level template code
    {% extends 'base.html' %}
  File "/mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/templates/base.html", line 52, in top-level template code
    {% block content %}
  File "/mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/templates/login.html", line 37, in block 'content'
    <a href="{{ url_for(request_password_reset) }}">Reset Here</a>
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py", line 1018, in url_for
    if endpoint[:1] == ".":
jinja2.exceptions.UndefinedError: 'request_password_reset' is undefined
2024-05-11 08:50:46,536 INFO: Tinker Full-stack App [in /mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/__init__.py:61]
2024-05-11 08:52:40,063 ERROR: Exception on /request-password-reset [GET] [in /home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py:825]
Traceback (most recent call last):
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/routes.py", line 41, in request_password_reset
    return render_template(
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/templating.py", line 152, in render_template
    return _render(app, template, context)
  File "/home/missmandazi/.local/lib/python3.8/site-packages/flask/templating.py", line 133, in _render
    rv = template.render(context)
  File "/home/missmandazi/.local/lib/python3.8/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/home/missmandazi/.local/lib/python3.8/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/templates/request_password_reset.html", line 1, in top-level template code
    {% extends 'base.html' %}
  File "/mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/templates/base.html", line 52, in top-level template code
    {% block content %}
  File "/mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/templates/request_password_reset.html", line 12, in block 'content'
    {{form.username.label}}<br>
  File "/home/missmandazi/.local/lib/python3.8/site-packages/jinja2/environment.py", line 485, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'app.forms.ResetPasswordRequestForm object' has no attribute 'username'
2024-05-11 08:55:26,828 INFO: Tinker Full-stack App [in /mnt/c/Users/Atuech Obosi/OneDrive/Desktop/software-development(Ayanda)/Chapter-13-Pagination-workthrough-/app/__init__.py:61]
